### PR TITLE
Made the issue stats query more secure with parameterized placeholders

### PR DESCRIPTION
**Pull request by tboerger**
Another try to get fine for https://github.com/gogits/gogs/issues/2892, but I need confirmation again that this works like expected.

### CODE REVIEW

**Comment by Unknwon** [comment]
I'll review before you confirm these changes do not break anything...

File: models/issue.go [hunk 1]
```diff
@@ -779,41 +779,48 @@ type IssueStatsOptions struct {
...
 	queryStr := "SELECT COUNT(*) FROM `issue` "
 	if opts.LabelID > 0 {
-		queryStr += "INNER JOIN `issue_label` ON `issue`.id=`issue_label`.issue_id AND `issue_label`.label_id=" + com.ToStr(opts.LabelID)
+		queryParams = append(queryParams, opts.LabelID)
+		queryStr += "INNER JOIN `issue_label` ON `issue`.id=`issue_label`.issue_id AND `issue_label`.label_id=?"
 	}
 
-	baseCond := " WHERE issue.repo_id=" + com.ToStr(opts.RepoID) + " AND issue.is_closed=?"
+	condParams = append(condParams, opts.RepoID)
+	baseCond := " WHERE issue.is_closed=? AND issue.repo_id=?"
```
Comment (andreynering) [inline models/issue.go]: Seems thet you missed to add the `is_closed` param to `condParams` above.

Comment (tboerger) [inline models/issue.go]: No, I have not missed it. It's a more or less dynamic parameter that gets set below in the switch. I just moved it to the beginning to be able to use the slice for the other parameters.

**Comment by tboerger** [comment]
@Unknwon I have tested it locally now. The query seems to be working like expected. I just realized something different, currently it's not possible to select multiple labels through the issues overview, should I create a followup pull request that makes this possible?

**Comment by tboerger** [comment]
Stop, need toi test more...

**Comment by tboerger** [comment]
And finally it works! @Unknwon if you know a more elegant way than defining an anonymous function for building a base query I'm open for it. Now the issue stats are calculated through XORM as well instead of a plain SQL query.

File: models/issue.go [hunk 2]
```diff
@@ -780,41 +780,64 @@ type IssueStatsOptions struct {
 func GetIssueStats(opts *IssueStatsOptions) *IssueStats {
 	stats := &IssueStats{}
 
-	queryStr := "SELECT COUNT(*) FROM `issue` "
-	if opts.LabelID > 0 {
-		queryStr += "INNER JOIN `issue_label` ON `issue`.id=`issue_label`.issue_id AND `issue_label`.label_id=" + com.ToStr(opts.LabelID)
-	}
+	countSession := func(repoID int64, isPull bool, labels string, milestoneID int64, assigneeID int64) *xorm.Session {
```
Comment (lunny) [inline models/issue.go]: Maybe 
```Go
countSession := func(opts * IssueStatsOptions) *xorm.Session {
...
}
```
is more clear.

**Comment by tboerger** [comment]
I have eleminated all possible vulns. Please review it.

**Comment by Unknwon** [comment]
@tboerger have you done all the tests?

**Comment by tboerger** [comment]
@Unknwon as far as I can see it works. Just needs your review :)

File: models/issue.go [hunk 3]
```diff
@@ -780,41 +780,64 @@ type IssueStatsOptions struct {
...
-	baseCond += " AND issue.is_pull=?"
 
 	switch opts.FilterMode {
 	case FM_ALL, FM_ASSIGN:
-		results, _ := x.Query(queryStr+baseCond, false, opts.IsPull)
-		stats.OpenCount = parseCountResult(results)
-		results, _ = x.Query(queryStr+baseCond, true, opts.IsPull)
-		stats.ClosedCount = parseCountResult(results)
-
+		openResult, _ := countSession(opts).
```
Comment (Unknwon) [inline models/issue.go]: Why not just `stats.OpenCount, _ = countSession...` here?

Comment (tboerger) [inline models/issue.go]: I can change it like that

Comment (tboerger) [inline models/issue.go]: Done, update the parts like that.

**Comment by tboerger** [comment]
@Unknwon I have applied your comment.

**Comment by Unknwon** [comment]
Thanks!

**Comment by Unknwon** [comment]
Hmm, I'm getting `0` of total count on dashboard issues tab.

**Comment by Unknwon** [comment]
Pushed a fix at least working now,.

**Pull request merged**